packages/components/src/components/typography/TruncateWithTooltip/TruncateWithTooltip.tsx

```diff
@@ -17,13 +17,19 @@ export const TruncateWithTooltip = ({ children, delayShow }: TruncateWithTooltip
     const containerRef = useRef(null);
 
     useEffect(() => {
-        if (containerRef.current) {
-            const { scrollWidth, clientWidth } = containerRef.current;
-
-            if (scrollWidth > clientWidth) {
+        if (!containerRef.current) return;
+        const { scrollWidth, scrollHeight } = containerRef.current;
+        const resizeObserver = new ResizeObserver(entries => {
+            const { inlineSize: width, blockSize: height } = entries[0].borderBoxSize?.[0];
+            if (scrollWidth > width || scrollHeight > height) {
                 setIsTooltipVisible(true);
+            } else {
+                setIsTooltipVisible(false);
```

Comment:
```suggestion
                setIsTooltipVisible(scrollWidth > width || scrollHeight > height);
```

And delete the  `if` ;)